### PR TITLE
Initial support for CHIP in NCS

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
WIP

### CODE REVIEW

File: samples/connectedhomeip/common/src/NFCWidget.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#pragma once
```
Comment (LuDuda) [inline samples/connectedhomeip/common/src/NFCWidget.h:7]: Doesn't Zephyr use ifdefs normally? We can leave it and wait for some comments.

File: samples/connectedhomeip/lock/src/BoltLockManager.cpp [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+bool BoltLockManager::IsUnlocked()
+{
+	return mState == State::UnlockingCompleted;
+}
+
+bool BoltLockManager::InitiateAction(Action action, bool chipInitiated)
+{
+	State newState = mState;
```
Comment (LuDuda) [inline samples/connectedhomeip/lock/src/BoltLockManager.cpp:41]: Unused variable.

File: samples/connectedhomeip/lock/src/gen/call-command-handler.cpp [hunk 3]
```diff
@@ -0,0 +1,109 @@
+/*
```
Comment (LuDuda) [inline samples/connectedhomeip/lock/src/gen/call-command-handler.cpp:1]: I wonder if we can have such license within our SDK. I would contact Carles and ask.

File: samples/connectedhomeip/lock/src/gen/endpoint_config.h [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *
+ *    Copyright (c) 2020 Silicon Labs
```
Comment (LuDuda) [inline samples/connectedhomeip/lock/src/gen/endpoint_config.h:20]: Even worse :smile:

**Review by LuDuda — COMMENTED** [review]
LGTM :+1: Small questions/requests.

File: samples/connectedhomeip/lock/src/gen/endpoint_config.h (hunk 4 above)
Comment (kkasperczyk-no) [reply samples/connectedhomeip/lock/src/gen/endpoint_config.h:20]: I'm not quite sure what to do with the files generated by ZAP. To be honest they are Silicon Labs property and I don't think that we should change the copyrights, as we are not modifying them in any way.

File: samples/connectedhomeip/lock/src/AppTask.cpp [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+#include "attribute-storage.h"
+#include "gen/cluster-id.h"
+
+#include <platform/CHIPDeviceLayer.h>
+
+#include <app/server/QRCodeUtil.h>
+#include <app/server/Server.h>
+#include <dk_buttons_and_leds.h>
+#include <logging/log.h>
+#include <setup_payload/QRCodeSetupPayloadGenerator.h>
```
Comment (kkasperczyk-no) [inline samples/connectedhomeip/lock/src/AppTask.cpp:26]: I think we don't need those two `setup_payload/...` headers.

File: samples/connectedhomeip/lock/src/BoltLockManager.h [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+	bool IsUnlocked();
+	bool IsActionInProgress();
+
+	bool InitiateAction(Action action, bool chipInitiated);
+	bool CompleteCurrentAction(bool &chipInitiated);
+
+private:
+	enum class State : uint8_t { LockingInitiated = 0, LockingCompleted, UnlockingInitiated, UnlockingCompleted };
+
+	static constexpr uint32_t ACTUATOR_MOVEMENT_PERIOS_MS = 2000;
```
Comment (kkasperczyk-no) [inline samples/connectedhomeip/lock/src/BoltLockManager.h:30]: Isn't there a rule to write const expressions using camelCase and MACRO_CASE is reserved for definitions?

Comment (Damian-Nordic) [reply samples/connectedhomeip/lock/src/BoltLockManager.h:30]: In CHIP some people use kConstantName and others CONSTANT_NAME. In nrf/samples consts are not distinguished anyhow and they're written as constant_name most of the time...

File: samples/connectedhomeip/lock/src/gen/endpoint_config.h (hunk 4 above)
Comment (Damian-Nordic) [reply samples/connectedhomeip/lock/src/gen/endpoint_config.h:20]: I think ZAP will eventually generate CHIP license headers, so probably we could use CHIP headers here, but the question is if it is OK with NCS code owners.

File: samples/connectedhomeip/common/src/NFCWidget.h (hunk 1 above)
Comment (Damian-Nordic) [reply samples/connectedhomeip/common/src/NFCWidget.h:7]: Yes, although I'm not sure the rule is written down anywhere. Should I change it?

File: samples/connectedhomeip/lock/src/BoltLockManager.h (hunk 6 above)
Comment (Damian-Nordic) [reply samples/connectedhomeip/lock/src/BoltLockManager.h:30]: on the other hand, enums in some samples follow the CONSTANT_NAME convention. Maybe I should leave ACTUATOR_MOVEMENT_PERIOD_MS and fix enums like State::LockingInitiated (should be State::LOCKING_INITIATED)?

Comment (kkasperczyk-no) [reply samples/connectedhomeip/lock/src/BoltLockManager.h:30]: In my opinion using camelCase looks better. I don't know if we should follow CHIP examples convention in NCS, as I've seen in some cases using local variables with rather pythonic style names like: `variable_name`, so it seems that there is no one strong naming convention in CHIP.

Comment (Damian-Nordic) [reply samples/connectedhomeip/lock/src/BoltLockManager.h:30]: ok, but I meant that enums in NCS samples use CONSTANT_NAME. Not sure if camelCase is a common convention in NCS. At least for constants. I guess that is something we may talk about during the final review...

File: samples/connectedhomeip/common/src/NFCWidget.h (hunk 1 above)
Comment (LuDuda) [reply samples/connectedhomeip/common/src/NFCWidget.h:7]: Let's wait for community comments.

File: samples/connectedhomeip/lock/src/gen/endpoint_config.h (hunk 4 above)
Comment (LuDuda) [reply samples/connectedhomeip/lock/src/gen/endpoint_config.h:20]: But as I understand this does not happen yet. So most-likely we would need to use Silicon Labs license as is.. Let's see what will be the comments from the NCS community.

**Pull request closed without merging**